src/libtiled/templatemanager.cpp

```diff
@@ -61,14 +65,34 @@ ObjectTemplate *TemplateManager::loadObjectTemplate(const QString &fileName, QSt
     if (!objectTemplate) {
         auto newTemplate = readObjectTemplate(fileName, error);
 
-        // This instance will not have an object. It is used to detect broken
-        // template references.
-        if (!newTemplate)
+        if (newTemplate) {
```

Comment:
Actually if the template failed to load, but the file exists anyway, we'll probably want to watch it too. Maybe just keep the code as-is and just add something like:

```c++
if (QFile::exists(fileName))
    mWatcher->addPath(fileName);
```